src/rpdk/core/contract/suite/handler_update.py

```diff
@@ -13,22 +13,20 @@
 
 @pytest.fixture(scope="module")
 def updated_resource(resource_client):
-    create_request = resource_client.generate_create_example()
+    create_request = model = resource_client.generate_create_example()
     try:
         _status, response, _error = resource_client.call_and_assert(
             Action.CREATE, OperationStatus.SUCCESS, create_request
         )
-        created_model = response["resourceModel"]
+        created_model = model = response["resourceModel"]
         update_request = resource_client.generate_update_example(created_model)
         _status, response, _error = resource_client.call_and_assert(
             Action.UPDATE, OperationStatus.SUCCESS, update_request, created_model
         )
         updated_model = response["resourceModel"]
```

Comment:
Do you need to also assign the model here as well so it is deleting the updated model if the update call is successful?

Reply:
we could but primary identifier shouldnt change between create and update and that should be included in the successful create.

Reply:
Thought about it more and there are cases where the delete operation depends on other properties besides the primary identifier so it would be better to assign the model here as well

Reply:
Thought about it more and there are cases where the delete operation depends on other properties besides the primary identifier so it would be better to assign the model here as well

Reply:
> Thought about it more and there are cases where the delete operation depends on other properties besides the primary identifier so it would be better to assign the model here as well

That would be in violation of the contract. Both Read and Delete handlers MUST operate on nothing more than the fields specified as primaryIdentifier in the schema. That's all they are guaranteed to receive.

Reply:
I don't think we ever specified that anywhere yet, but you are right we need to define a contract for handler input